File: cpp/1732.0_Find_the_Highest_Altitude.cpp
/*
执行用时：0 ms, 在所有 C++ 提交中击败了100.00% 的用户
内存消耗：7.7 MB, 在所有 C++ 提交中击败了87.14% 的用户
通过测试用例：80 / 80
*/
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int highest = 0, altitude = 0;
        for (auto &g : gain) {
            altitude += g;
            highest = max(highest, altitude);
        }
        return highest;
    }
};

File: problems/1732.0_Find_the_Highest_Altitude.py
'''
Runtime: 51 ms, faster than 54.67% of Python3 online submissions for Find the Highest Altitude.
Memory Usage: 16.3 MB, less than 26.06% of Python3 online submissions for Find the Highest Altitude.
'''
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        ans = altitude = 0
        for g in gain:
            altitude += g
            ans = max(ans, altitude)
        return ans

'''
Runtime: 57 ms, faster than 16.50% of Python3 online submissions for Find the Highest Altitude.
Memory Usage: 16.1 MB, less than 91.15% of Python3 online submissions for Find the Highest Altitude.
'''
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        return max(accumulate(gain, initial=0))
        